Source/Library/Board.cpp

```diff
@@ -3,12 +3,62 @@
 using namespace TicTacToe;
 
 Board::Token Board::get_piece(int x, int y) const {
-  return Token::NONE;
+  if(x < 0 || y < 0 || x >= ROW_COUNT || y >= COLUMN_COUNT)
+    throw std::out_of_range(
+      "The x or y coordinate is outside of the board's range.");
+
+  return board_[x][y];
 }
 
-void Board::set_piece(int x, int y, Token token) {}
+void Board::set_piece(int x, int y, Token token) {
+  if(x < 0 || y < 0 || x >= ROW_COUNT || y >= COLUMN_COUNT)
+    throw std::out_of_range(
+      "The x or y coordinate is outside of the board's range.");
+
+  board_[x][y] = token;
+}
 
 
 Board::Token TicTacToe::evaluate_winner(const Board& board) {
+  /** Count of each row */
+  std::vector<int> rows(Board::ROW_COUNT, 0);
+
+  /** Count of each column */
+  std::vector<int> cols(Board::COLUMN_COUNT, 0);
+
+  /** Count on the diagonal */
+  auto diag = 0;
+
+  /** Count on the reverse diagonal */
+  auto reverse_diag = 0;
+
+  for(auto x = 0; x < Board::ROW_COUNT; ++x) {
+    for(auto y = 0; y < Board::COLUMN_COUNT; ++y) {
+      auto piece = board.get_piece(x, y);
+      auto point = (piece == Board::Token::NONE ? 0 : 
```

Comment:
Ternary operator is not allowed, use IIFE:

    auto point = [&] {
      if(piece == Board::Token::NONE) {
        return 0;
      } else if(piece == Board::Token::X) {
        return 1;
      } else {
        return -1;
      }
    }();